### PR TITLE
adding in job source logic to coalesce job calculation and enhancing …

### PR DESCRIPTION
**Pull request by Catherine-Q-MT**
…tests

# Description
once we have added the new column to our dataset we need to start adding the logic for populating the source. 

# How to test
Unit tests should pass and contain assertion that test the contents of the job source column, these should then be visible in Athena if we run the prepare locations job

# Developer checklist
- [ ] Unit test
- [ ] Linked to Trello ticket
- [ ] Documentation up to date

### CODE REVIEW

**Review by garypolzin — APPROVED** [review]
One potential idea but feel free to disregard

File: tests/unit/test_calculate_job_coalesce_total_staff_worker_records.py [hunk 1]
```diff
@@ -35,16 +36,15 @@ def setUp(self):
...
         df = calculate_jobcount_coalesce_totalstaff_wkrrecs(df)
         self.assertEqual(df.count(), 3)
-
         df = df.collect()
-        self.assertEqual(df[0]["job_count"], 20)
-        self.assertEqual(df[1]["job_count"], 30)
-        self.assertEqual(df[2]["job_count"], None)
+        self.assertEqual(df[0]["job_count"], 20, "coalesce_total_staff_wkrrecs")
+        self.assertEqual(df[1]["job_count"], 30, "coalesce_total_staff_wkrrecs")
+        self.assertEqual(df[2]["job_count"], None, None)
```
Comment (garypolzin) [inline tests/unit/test_calculate_job_coalesce_total_staff_worker_records.py:50]: these tests aren't actually testing the source text, this is testing that the first two objects are equal, the third object isn't part of the tests

you'll need two tests for each, one for the job_count column and one for the job_count_source
self.assertEqual(df[0]["job_count"], 20)
self.assertEqual(df[0]["job_count_source"], "coalesce_total_staff_wkrrecs")

**Review by garypolzin — CHANGES_REQUESTED** [review]

File: utils/prepare_locations_utils/job_calculator/calculate_jobcount_coalesce_totalstaff_wkrrecs.py [hunk 2]
```diff
@@ -26,8 +25,24 @@ def calculate_jobcount_coalesce_totalstaff_wkrrecs(input_df):
...
+            F.when(
+                F.col("job_count_temp").isNotNull(), F.col("job_count_temp")
+            ).otherwise(F.coalesce(F.col("job_count")))
+        ),
     )
+    output_df = df_with_job_count_populated.drop("job_count_temp")
+    return output_df
 
 
 def select_the_non_null_value_of_total_staff_and_worker_record_count(input_df):
```
[the remark this answers is not in the document]
Comment (Catherine-Q-MT) [reply utils/prepare_locations_utils/job_calculator/calculate_jobcount_coalesce_totalstaff_wkrrecs.py:48]: great idea. will change

File: tests/unit/test_calculate_job_coalesce_total_staff_worker_records.py (hunk 1 above)
Comment (Catherine-Q-MT) [reply tests/unit/test_calculate_job_coalesce_total_staff_worker_records.py:50]: As discussed, how github is displaying this is confusing. in split mode we can see its written as you specified:
image

Comment (Catherine-Q-MT) [reply tests/unit/test_calculate_job_coalesce_total_staff_worker_records.py:50]: thanks! will correct

**Pull request closed without merging**